lib/model-builder.js

```diff
@@ -257,10 +257,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
       delete properties[p];
     }
 
-    // Warn about properties with unsupported names
+    // Throw error for properties with unsupported names
     if (/\./.test(p)) {
-      deprecated('Property names containing a dot are not supported. ' +
-        'Model: ' + className + ', property: ' + p);
+      throw new Error('Property names containing dot(s) are not supported: ' + p);
```

Comment:
Please include the model name (`className`) in the message, otherwise it's difficult to find the model which defines such invalid property.